packages/runtime/src/runtime.ts

```diff
@@ -814,9 +816,10 @@ export class Runtime {
 
     const txnParameters = Array.isArray(txnParams) ? txnParams : [txnParams];
     // Run TEAL program associated with each transaction without interacting with store.
-    for (const txnParam of txnParameters) {
+    for (const [index, txnParam] of txnParameters.entries()) {
       this.assertAmbiguousTxnParams(txnParam);
       if (txnParam.sign === SignType.LogicSignature) {
+        this.ctx.tx = this.ctx.gtxs[index]; // update current tx
```

Comment:
can you please explain this ?

Reply:
should we set this.ctx.tx back to previous after we are done with lsig ?

Reply:
> can you please explain this ?

We need to set the current transaction (for correct `GroupIndex`) before executing stateless TEAL

> should we set this.ctx.tx back to previous after we are done with lsig ?

Not sure if setting previous config the right way. What if the first tx is the stateless one?

Reply:
for the ssc `tx` is first transaction in `gtxs`, right ?

Reply:
> should we set this.ctx.tx back to previous after we are done with lsig ?

Doing this.